### DISCUSSION TITLE
Welcome to viper Discussions!

### DISCUSSION THREAD

**Discussion by spf13** [spf13/viper#1051] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lr638638** [comment]
My programs is running in docker (Docker version 19.03.5, build 633a0ea838)
1. docker run -it -v /etc/config.json:/etc/config/config.json
2. have add the code as bellow:
c.Viper.OnConfigChange(func(e fsnotify.Event) {
		fmt.Println("abc:OnConfigChange")
		c.Update()
	})
	c.Viper.WatchConfig()

3. modify /etc/config.json in local host
4. can't print  log "abc:OnConfigChange"
5. modify the codes in viper.go:498
                watcher.Add(configDir)
		watcher.Add(configFile) //add by me
6. Do step 1 and 3 again
7. print the log "abc:OnConfigChange"

It's have bug when the config file volume mount in docker.
